jerry-ext/args/args.c

```diff
@@ -0,0 +1,97 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/**
+ * It validate the JS arguments and assign them to the native argument.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerryx_validate_and_assign_args (const jerry_value_t *js_arg_p, /**< points to the js argument */
+                                 const jerry_length_t js_arg_cnt, /**< the count of the js arguments */
+                                 const jerryx_arg_t *c_arg_p, /**< points to the jerryx_arg_t */
+                                 const jerry_length_t c_arg_cnt) /**< the count of the native argument */
+{
+  jerry_value_t ret = jerry_create_undefined ();
+
+  if (c_arg_cnt == 0)
+  {
+    return ret;
+  }
+
+  jerryx_arg_iterator_t iterator =
```

Comment:
@akosthekiss 2 other very simple and common use case for which iterator would be useful:

1. I'd like to add more specific error messages at some point, i.e. `Expected boolean at argument with index 2`.

2. I've seen projects where they check that the number of arguments matches the number of expected arguments. This can be done using a "terminator transform", that would be added as the last `jerryx_arg_t`. The transformer checks whether exactly N input arguments have been processed, or throws an error otherwise.

To implement these use cases, you would need the index of the current argument in the transformer (`iterator_p->js_arg_index`). It's easy to add a helper function `int jerryx_arg_iterator_get_arg_index(jerryx_arg_iterator_t  const *iterator_p)`. It's not so easy to have to change the function signature once it's part of the public API and people start to rely on it.